node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -363,17 +363,41 @@ object P2PClient extends P2PLogger {
       if (remainingBytes.length <= 0) {
         (accum.reverse, remainingBytes)
       } else {
-        val messageTry = Try(NetworkMessage(remainingBytes))
-        messageTry match {
-          case Success(message) =>
+        val headerTry = Try(
+          NetworkHeader.fromBytes(remainingBytes.take(NetworkHeader.bytesSize)))
+        headerTry match {
+          case Success(header) =>
+            val payloadBytes = remainingBytes
+              .drop(NetworkHeader.bytesSize)
+              .take(header.payloadSize.toInt)
+
             val newRemainingBytes =
-              remainingBytes.slice(message.bytes.length, remainingBytes.length)
-            logger.trace(
-              s"Parsed a message=${message.header.commandName} from bytes, continuing with remainingBytes=${newRemainingBytes.length}")
-            loop(newRemainingBytes, message :: accum)
+              remainingBytes.drop(NetworkHeader.bytesSize + payloadBytes.size)
+
+            // If it's a message type we know, try to parse it
+            if (NetworkPayload.commandNames.contains(header.commandName)) {
+              Try(NetworkMessage(header.bytes ++ payloadBytes)) match {
+                case Success(message) =>
+                  logger.trace(
+                    s"Parsed a message=${message.header.commandName} from bytes, continuing with remainingBytes=${newRemainingBytes.length}")
+
+                  loop(newRemainingBytes, message :: accum)
+                case Failure(_) =>
+                  // Can't parse message yet, we need to wait for more bytes
+                  (accum.reverse, remainingBytes)
```

Comment:
Why `accum.reverse`?

Reply:
Oh I see, never mind

Reply:
Although  if we change `accum` type to `Vector` we could simply append new messages, and wouldn't need `reverse`

Reply:
Changed to use `Vector` and removed the `reverse`s